src/solvers/strings/string_refinement.cpp

```
@@ -1804,6 +1819,140 @@ exprt substitute_array_lists(exprt expr, size_t string_max_length)
   return expr;
 }
 
+/// Given an index set, return a sorted vector of the concrete indices
+/// defined in the current model.
+/// \param index_set index set
+/// \return concrete indices
+std::vector<size_t> string_refinementt::get_model_defined_indices(
+  const std::set<exprt> &index_set) const
+{
+  std::vector<size_t> defined_indices;
+  for(const auto &index_expr : index_set)
+  {
+    if(auto index = numeric_cast<size_t>(simplify_expr(get(index_expr), ns)))
```

I'm a bit worried that this will silently ignore expression that cannot be simplified to a constant